exonum/src/events/network.rs

```diff
@@ -290,31 +315,30 @@ impl RequestHandler {
             .map_err(|_| other_error("no network requests"))
             .for_each(move |request| {
                 match request {
+                    NetworkRequest::ConnectToPeer(peer) => {
+                        let conn_tx = outgoing_connections.get_or_create(
+                            peer,
+                            &network_tx,
+                            &handle,
+                            network_config,
+                            &handshake_params,
+                        );
+
+                        if let Some(conn_tx) = conn_tx {
+                            to_box(conn_tx)
+                        } else {
+                            to_box(Self::send_unable_connect_event(peer, &network_tx))
```

Comment:
Does `send_unable_connect_event` really need to return something if the result is discarded anyway in all cases?

Reply:
Yes, `network_tx` is unsync sender and needs to be polled.

Reply:
Sorry for being obscure; I've meant something like

```rust
fn send_unable_connect_event(
    peer: SocketAddr,
    network_tx: mpsc::Sender<NetworkEvent>,
) -> Box<dyn Future<Item = (), Error = io::Error>> {
    let event = NetworkEvent::UnableConnectToPeer(peer);
    Box::new(
        network_tx
            .send(event)
            .map(drop)
            .map_err(|_| other_error("can't send network event")),
    )
}
```

Note `map(drop)`; we don't use the sender after we return it from the future.